### PR TITLE
Vin/mags wrapup

### PR DESCRIPTION
**Pull request by vinnie4k**
## Overview

This PR finalizes magazines MVP to prepare for shipping.

### Magazine Scrollbar

- Created `MagsScrollbarView` to resolve the buggy scrollbar issue mentioned in PR #118
- Tapping on the table of contents menu fades the scrollbar and page indicator into/out of the view (hidden by default)
- Made changes to `PDFKitView`, `MagazineReaderView`, and `MagazineCell` to accommodate

### Magazine Reader

- Created `PageIndicatorView` to keep track of the current page
    - Synchronized with both page swiping and dragging the scrollbar using Notifications
    - Can be toggled on/off using the table of contents menu

### Minor Changes

- Moved `PublicationDetail` folder outside of `Onboarding` folder
- Removed NSFW blur as requested
- Fixed bug where publication view is closed after opening
- Made search bar scroll with list in home screen

## Test Coverage

- Used different devices with varying screen sizes in Simulator to see if there were any UI issues
- In-depth playtesting to minimize any bugs that may be present

## Related PRs or Issues

 - (RESOLVED) Tapping on “See more” in the toolbar in `MagazineReaderView` loads the `PublicationContentView` but quickly closes out. Issue has something to do with the property `pdfView` initialized in `MagazineReaderView`, but feature won’t work if removed.

## Screenshots

<details>

  <summary>Mags Scrollbar/Page Indicator</summary>

  <video src="https://user-images.githubusercontent.com/75594943/223290709-7490f267-f465-4657-b20b-91b972265288.mp4" >

</details>

### CODE REVIEW

File: Volume/Supporting Views/MagsScrollbarView.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+// MARK: - Constants
+extension MagsScrollbarView {
+
+    private struct Constants {
+        static let thumbnailWidth: CGFloat = 20
+        static let thumbnailHeight: CGFloat = 40
+        static let scrollbarColor: UIColor = UIColor(Color(white: 0.93))
+    }
```
Comment (ReadeP) [inline Volume/Supporting Views/MagsScrollbarView.swift:37]: add newline after 37

File: Volume/Supporting Views/PageIndicatorView.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+// MARK: - Constants
+extension PageIndicatorView {
+
+    private struct Constants {
+        static let indicatorWidth: CGFloat = 80
+        static let indicatorHeight: CGFloat = 30
+        static let indicatorCorner: CGFloat = 15
+        static let indicatorColor = Color(white: 0.93)
+    }
```
Comment (ReadeP) [inline Volume/Supporting Views/PageIndicatorView.swift:46]: newline after 46

File: Volume/View Models/ArticleInfo.swift [hunk 3]
```diff
@@ -29,7 +29,7 @@ struct ArticleInfo: View {
                     .font(.helveticaNeueMedium(size: isDebrief ? 24 : 16))
                     .lineLimit(3)
                     .padding(.top, 0.5)
-                    .blur(radius: article.isNsfw ? 3 : 0)
+//                    .blur(radius: article.isNsfw ? 3 : 0) Remove NSFW Blur
```
Comment (ReadeP) [inline Volume/View Models/ArticleInfo.swift:32]: Can we delete this if we don't need it?

File: Volume/Views/MainView/MagazineReaderView.swift [hunk 4]
```diff
@@ -113,12 +122,26 @@ struct MagazineReaderView: View {
...
                 if let pdfDoc = magazine?.pdfDoc {
-                    PDFKitView(pdfDoc: pdfDoc)
+                    PDFKitView(pdfView: pdfView, pdfDoc: pdfDoc)
+                        .overlay(showScrollbar
+                                 ? PageIndicatorView(
+                                        totalPage: pdfDoc.pageCount,
+                                        pdfView: pdfView
+                                    ).padding([.top, .trailing])
+                                 : nil
+                                 ,alignment: .topTrailing)
```
Comment (ReadeP) [inline Volume/Views/MainView/MagazineReaderView.swift:132]: This looks a little difficult to read

**Review by ReadeP — APPROVED** [review]
Great work on your first PR Vin! I left some feedback so please address it before you merge! 🤩

File: Volume/View Models/MagazineCell.swift [hunk 5]
```diff
@@ -38,9 +38,9 @@ struct MagazineCell: View {
         VStack(alignment: .leading) {
             Group {
                 if let pdfDoc = magazine.pdfDoc {
-                    PDFKitView(pdfDoc: pdfDoc, isCover: true)
+                    PDFKitView(pdfView: PDFViewUnselectable(), pdfDoc: pdfDoc, isCover: true)
                 } else {
-                    PDFKitView(pdfDoc: PDFDocument(), isCover: true)
+                    PDFKitView(pdfView: PDFViewUnselectable(), pdfDoc: PDFDocument(), isCover: true)
                 }
```
Comment (dishanest) [inline Volume/View Models/MagazineCell.swift:44]: lowkey could condense this to 
```swift
PDFKitView(pdfView: PDFViewUnselectable(), pdfDoc: magazine.pdfDoc ?? PDFDocument(), isCover: true)
```

**Review by dishanest — DISMISSED** [review]
Amazing first PR! Just a few small cases I think you should handle and you should be set.

File: Volume/Supporting Views/PDFKitView.swift [hunk 6]
```diff
@@ -12,18 +12,23 @@ import PDFKit
...
         pdfView.autoScales = true
         pdfView.usePageViewController(true)
         pdfView.displayDirection = isCover ? .vertical : .horizontal
         pdfView.displaysAsBook = true
         pdfView.displaysPageBreaks = false
+        pdfView.displayMode = .singlePage
 
+        let scrollView = pdfView.subviews.first?.subviews.first as? UIScrollView
+        scrollView?.showsVerticalScrollIndicator = false
+        scrollView?.showsHorizontalScrollIndicator = false
```
[the remark this answers is not in the document]
Comment (vinnie4k) [reply Volume/Supporting Views/PDFKitView.swift:30]: The first subview of a PDFView is always a UIScrollView.

**Pull request merged**